#!/usr/bin/env python
PACKAGE = "pd_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

#        NOMBRE        TIPO   FLAG         DESCRIPTION        DEFAULT  MIN  MAX 
# Example
# gen.add("position_kp", double_t, 0, "Position controller kp gain",  1.04,0, 50)

gen.add("Kp_x", double_t, 0, "Position controller Kp gain",   5.0  ,0, 50)
gen.add("Kp_y", double_t, 0, "Position controller Kp gain",   5.0  ,0, 50)
gen.add("Kp_z", double_t, 0, "Position controller Kp gain",   6.0  ,0, 50)

gen.add("Kd_x", double_t, 0, "Position controller Kd gain",   3.0  ,0, 50)
gen.add("Kd_y", double_t, 0, "Position controller Kd gain",   3.0  ,0, 50)
gen.add("Kd_z", double_t, 0, "Position controller Kd gain",   3.0  ,0, 50)

gen.add("Ki_x", double_t, 0, "Position controller Ki gain",   0.001  ,0, 1)
gen.add("Ki_y", double_t, 0, "Position controller Ki gain",   0.001  ,0, 1)
gen.add("Ki_z", double_t, 0, "Position controller Ki gain",   0.001  ,0, 1)

gen.add("Kp_roll", double_t, 0,  "Angle  controller Kp gain",   5.5  ,0, 50)
gen.add("Kp_pitch", double_t, 0, "Angle  controller Kp gain",   5.5  ,0, 50)
gen.add("Kp_yaw", double_t, 0,   "Angle  controller Kp gain",   5  ,0, 50)


exit(gen.generate(PACKAGE, "pd_controller", "Controller"))